sw/device/lib/testing/mock_mmio.h

```diff
@@ -81,131 +173,285 @@ class MmioTest {
+ * |MmioTest|.
  *
  * This expectation is sequenced with all other |EXPECT_READ| and |EXPECT_WRITE|
  * calls.
  */
-#define EXPECT_READ32(offset, value) \
-  EXPECT_READ32_AT(this->dev(), offset, value)
+#define EXPECT_READ32(offset, ...) \
+  EXPECT_READ32_AT(this->dev(), offset, __VA_ARGS__)
 
 /**
  * Expect a write to the given offset with the given 8-bit value.
  *
+ * The value may be given as an integer, a pointer to little-endian data,
+ * or a |std::initializer_list<BitField>|.
+ *
  * This function is only available in tests using a fixture that derives
- * |DeviceTest|.
+ * |MmioTest|.
  *
  * This expectation is sequenced with all other |EXPECT_READ| and |EXPECT_WRITE|
  * calls.
  */
-#define EXPECT_WRITE8(offset, value) \
-  EXPECT_WRITE8_AT(this->dev(), offset, value);
+#define EXPECT_WRITE8(offset, ...) \
+  EXPECT_WRITE8_AT(this->dev(), offset, __VA_ARGS__);
 
 /**
  * Expect a write to the given offset with the given 16-bit value.
  *
+ * The value may be given as an integer, a pointer to little-endian data,
+ * or a |std::initializer_list<BitField>|.
+ *
  * This function is only available in tests using a fixture that derives
- * |DeviceTest|.
+ * |MmioTest|.
  *
  * This expectation is sequenced with all other |EXPECT_READ| and |EXPECT_WRITE|
  * calls.
  */
-#define EXPECT_WRITE16(offset, value) \
-  EXPECT_WRITE16_AT(this->dev(), offset, value);
+#define EXPECT_WRITE16(offset, ...) \
+  EXPECT_WRITE16_AT(this->dev(), offset, __VA_ARGS__);
 
 /**
  * Expect a write to the given offset with the given 32-bit value.
  *
+ * The value may be given as an integer, a pointer to little-endian data,
+ * or a |std::initializer_list<BitField>|.
+ *
+ * This function is only available in tests using a fixture that derives
+ * |MmioTest|.
+ *
+ * This expectation is sequenced with all other |EXPECT_READ| and |EXPECT_WRITE|
+ * calls.
+ */
+#define EXPECT_WRITE32(offset, ...) \
+  EXPECT_WRITE32_AT(this->dev(), offset, __VA_ARGS__);
+
+#define EXPECT_MASK_INTERAL_(width, dev, off, ...)                         \
+  do {                                                                     \
+    auto &device = dev;                                                    \
+    std::initializer_list<mock_mmio::MaskedBitField> fields = __VA_ARGS__; \
+                                                                           \
+    using Int = uint##width##_t;                                           \
+    auto val = device.GarbageMemory<Int>();                                \
+    EXPECT_READ##width##_AT(device, off, val);                             \
+                                                                           \
+    for (auto field : fields) {                                            \
+      ASSERT_LT(field.offset, sizeof(Int) * 8);                            \
+      ASSERT_LE(field.mask, std::numeric_limits<Int>::max());              \
+                                                                           \
```

Comment:
Maybe add a sanity check to ensure that the field value is not bigger than the field mask.
`ASSERT_LE(field.value, field.mask);`